app/src/main/java/de/grobox/transportr/trips/search/ProductDialogFragment.java

```diff
@@ -114,6 +130,12 @@ public void onStart() {
 		}
 	}
 
+	@Override
+	public void onAttach(@NonNull @NotNull Context context) {
+		super.onAttach(context);
+		((TransportrApplication) getActivity().getApplicationContext()).getComponent().inject(this);
```

Comment:
i am not sure how this fragment did work previously without calling dagger `inject` and why the issue appeared in my PR.
however, this solves the NullRef you mentioned @Altonss